components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -1758,6 +1773,48 @@ private void parsePersistAccessTokenAliasConfig(OMElement oauthConfigElem) {
         }
     }
 
+    private void parseRetainOldAccessTokensConfig(OMElement oauthCleanupConfigElem) {
+
+        OMElement tokenCleanElem = oauthCleanupConfigElem.getFirstChildWithName(getQNameWithIdentityNS(ConfigElements.OAUTH2_TOKEN_CLEAN_ELEM));
+        if (tokenCleanElem != null) {
+            OMElement oldTokenRetainConfigElem = tokenCleanElem.getFirstChildWithName(getQNameWithIdentityNS(ConfigElements.RETAIN_OLD_ACCESS_TOKENS));
+            if (oldTokenRetainConfigElem != null && !"".equals(oldTokenRetainConfigElem.getText().trim())) {
+                retainOldAccessTokens = oldTokenRetainConfigElem.getText().trim();
+                if (log.isDebugEnabled()) {
+                    log.debug("Retain old access token is set to : " + retainOldAccessTokens);
+                }
+            } else {
+                retainOldAccessTokens = "false";
+                if (log.isDebugEnabled()) {
+                    log.debug("Retain old access token  is not defined.Default config will be used");
+                }
+            }
+        } else {
+            tokenCleanupFeatureEnable = "false";
+        }
+    }
+
+    private void tokenCleanupFeatureConfig(OMElement oauthCleanupConfigElem) {
+
+        OMElement tokenCleanElem = oauthCleanupConfigElem.getFirstChildWithName(getQNameWithIdentityNS(ConfigElements.OAUTH2_TOKEN_CLEAN_ELEM));
+        if (tokenCleanElem != null) {
+            OMElement tokenCleanupConfigElem = tokenCleanElem.getFirstChildWithName(getQNameWithIdentityNS(ConfigElements.TOKEN_CLEANUP_FEATURE));
+            if (tokenCleanupConfigElem != null && !"".equals(tokenCleanupConfigElem.getText().trim())) {
```

Comment:
Check tokenCleanupConfigElem.getText() != null
Can't we use isEmpty() method in apache utils instead  !"".equals?
"tokenCleanupConfigElem.getText().trim().isEmpty()"

Reply:
fixed in 54d6584